examples/datasync-plugin/main.go

```diff
@@ -28,65 +28,28 @@ import (
 // ************************************************************************/
 
 // PluginName represents name of plugin.
-const PluginName = "example"
+const PluginName = "datasync-example"
 
 func main() {
-	// Start Agent with ExamplePlugin, ETCDPlugin & FlavorLocal (reused cn-infra plugins).
-	/*agent := local.NewAgent(local.WithPlugins(func(flavor *local.FlavorLocal) []*core.NamedPlugin {
-		etcdPlug := &etcd.Plugin{}
-		etcdDataSync := &kvdbsync.Plugin{}
-		resyncOrch := &resync.Plugin{}
-
-		etcdPlug.Deps.PluginInfraDeps = *flavor.InfraDeps("etcd", local.WithConf())
-		resyncOrch.Deps.PluginLogDeps = *flavor.LogDeps("etcd-resync")
-		connectors.InjectKVDBSync(etcdDataSync, etcdPlug, etcdPlug.PluginName, flavor, resyncOrch)
-
-		examplePlug := &ExamplePlugin{closeChannel: exampleFinished}
-		examplePlug.Deps.PluginInfraDeps = *flavor.InfraDeps("etcd-example")
-		examplePlug.Deps.Publisher = etcdDataSync // Inject datasync Watcher to example plugin.
-		examplePlug.Deps.Watcher = etcdDataSync   // Inject datasync Publisher to example plugin.
-
-		return []*core.NamedPlugin{
-			{etcdPlug.PluginName, etcdPlug},
-			{etcdDataSync.PluginName, etcdDataSync},
-			{resyncOrch.PluginName, resyncOrch},
-			{examplePlug.PluginName, examplePlug},
-		}
+	// Prepare ETCD data sync plugin as an plugin dependency
+	etcdDs := kvdbsync.NewPlugin(kvdbsync.UseDeps(func(deps *kvdbsync.Deps) {
+		deps.KvPlugin = &etcd.DefaultPlugin
+		deps.ResyncOrch = &resync.DefaultPlugin
```

Comment:
The resync was recently added as default dependency, since datasync is only ever used with it. You don't have to specifically set it. For the KV store, there is special option `UseKV` that makes all of this much shorter and cleaner.